packages/myst-to-react/src/proof.tsx

```diff
@@ -0,0 +1,213 @@
+  const classes =
+    className
+      ?.split(' ')
+      .map((s) => s.trim().toLowerCase())
+      .filter((s) => !!s) ?? [];
+  return [...new Set(classes)];
+}
+
+function capitalize(kind?: string) {
+  if (!kind) return '';
+  return kind.slice(0, 1).toUpperCase() + kind.slice(1);
+}
+
+function getColor({ kind }: { kind?: ProofKind | string; classes?: string[] }): {
+  color: Color;
+} {
+  switch (kind) {
+    case ProofKind.proof:
+    case ProofKind.algorithm:
+      return { color: 'gray' };
+    case ProofKind.lemma:
+    case ProofKind.conjecture:
+    case ProofKind.theorem:
+      return { color: 'purple' };
+    case ProofKind.observation:
+    case ProofKind.assumption:
+    case ProofKind.axiom:
+      return { color: 'yellow' };
+    case ProofKind.criterion:
+    case ProofKind.corollary:
+    case ProofKind.property:
+      return { color: 'orange' };
+    case ProofKind.example:
+      return { color: 'green' };
+    case ProofKind.remark:
+      return { color: 'red' };
+    case ProofKind.definition:
+    case ProofKind.proposition:
+    default:
+      return { color: 'blue' };
+  }
+}
+
+const WrapperElement = ({
+  id,
+  dropdown,
+  className,
+  children,
+}: {
+  id?: string;
+  className?: string;
+  children: React.ReactNode;
+  dropdown?: boolean;
+}) => {
+  if (dropdown)
+    return (
+      <details id={id} className={className}>
+        {children}
+      </details>
+    );
+  return (
+    <aside id={id} className={className}>
+      {children}
+    </aside>
+  );
+};
+
+const HeaderElement = ({
+  dropdown,
+  className,
+  children,
+}: {
+  className?: string;
+  children: React.ReactNode;
+  dropdown?: boolean;
+}) => {
+  if (dropdown) return <summary className={className}>{children}</summary>;
+  return <div className={className}>{children}</div>;
+};
+
+const iconClass = 'h-8 w-8 inline-block pl-2 mr-2 self-center flex-none';
+
+export function Proof({
+  title,
+  kind,
+  color,
+  dropdown,
+  children,
+  identifier,
+  enumerator,
+}: {
+  title?: React.ReactNode;
+  color?: Color;
+  kind?: ProofKind;
+  children: React.ReactNode;
+  dropdown?: boolean;
+  identifier?: string;
+  enumerator?: string;
+}) {
+  return (
+    <WrapperElement
+      id={identifier}
+      dropdown={dropdown}
+      className={classNames('my-5 shadow', 'dark:bg-stone-800', 'overflow-hidden', {
+        // 'shadow-lg': !color || color === 'gray',
+        // 'border-l-2 border-blue-500': color === 'blue',
+        // 'border-l-2 border-green-500': color === 'green',
+        // 'border-l-2 border-amber-500': color === 'yellow',
+        // 'border-l-2 border-orange-500': color === 'orange',
+        // 'border-l-2 border-red-500': color === 'red',
+      })}
+    >
+      <HeaderElement
+        dropdown={dropdown}
+        className={classNames(
+          'm-0 font-medium py-2 flex min-w-0',
+          'text-md',
+          '',
+          'border-y dark:border-y-0 dark:border-l-4 border-slate-400',
```

Comment:
So (this is great btw) the `dark:border-l-4` here causes a small misalignment. To fix it either `dark:ml-3` is needed on L164 below, or else `dark:border-l-4` should be moved up to the WrapperElement. I prefer the latter solution for parity with dark admonitions.

I tried to do it myself but it results in the wrong border color and I'm not sure how to fix that.

Reply:
Ah, ok -- first attempt is in #90, I misread this comment though and see now you were thinking proofs should be more like admonitions. I went back and forth on this in my first pass and decided to make them a bit different intentionally, but not a lot of reason behind that.

I will explore this now and post a screenshot of options.

Reply:
Ok, I think I do like this better!

![image](https://github.com/executablebooks/myst-theme/assets/913249/d8147d7a-15cf-44b6-92a3-24f80376f8ef)

Reply:
I like it better too, I think -- meaning w/ the long border. I am not sure whether the original border color was nicer... though not a big deal.